open-apparel-registry/open-apparel-registry
# Add django-rest-auth and enable signup, login, and logout

## Overview

This PR adds django-rest-auth and wires up the frontend to enable users to sign up, login, and logout.

Along the way I replaced the CORS setup with a create-react-app proxy and also renamed the `ShowOnly` component's `if` prop to `showChildren`, which avoids a conflict with the JS `if` keyword.

I also added an updated 0001 migration in order to use email addresses as usernames.

Connects #70
Connects #46 
Connects #31 

## Demo

![register](https://user-images.githubusercontent.com/4165523/51276338-6fa52100-19a2-11e9-819b-57817b841c3d.gif)
![login-error](https://user-images.githubusercontent.com/4165523/51276339-6fa52100-19a2-11e9-859c-19e3168e3d07.gif)
![login](https://user-images.githubusercontent.com/4165523/51276341-703db780-19a2-11e9-90b9-87c50228a7d7.gif)

## Testing Instructions

- vagrant destroy your VM
- `./scripts/setup` and verify that it concludes successfully
- `./scripts/server` then visit the app in the browser on port 6543
- click on the "Register" link to register, then try signing up for an account. Verify that:
    - all required fields have validation messages about being required
    - mismsatching passwords also show a validation message
    - the form can successfully be submitted and creates a new user
- after your user saves, click log out
- click on the "Login" link and enter your username and password
- sign in and verify that you are redirected to the placeholder profile page and see the navbar login state change
- refresh the page and verify that you are signed in since you have an active session
- run `./scripts/test` and verify that the linter and tests pass
